addons/account/models/account_move.py

```diff
@@ -3698,7 +3702,10 @@ def _post(self, soft=True):
             if move.state in ['posted', 'cancel']:
                 raise UserError(_('The entry %s (id %s) must be in draft.', move.name, move.id))
             if not move.line_ids.filtered(lambda line: line.display_type not in ('line_section', 'line_note')):
-                raise UserError(_('You need to add a line before posting.'))
+                raise UserError(_(
+                    "Make sure you add a line before posting the invoice. Invoices can't be created from thin air.\n"
+                    "You’ll thank me later for avoiding a weird explanation to your auditor :)"
```

Comment:
Mixed apostrophes: `'` and `’`. I'd advocate picking either one and sticking with it